File: fs-project-backend/src/main/java/com/example/fs_project_backend/service/UserTypeService.java
package com.example.fs_project_backend.service;

import com.example.fs_project_backend.entity.UserType;

import java.util.List;

public interface UserTypeService {

    List<UserType> getAll();

    UserType create(UserType userType);

    UserType update(Integer id, UserType updated);

    void delete(Integer id);
}

File: fs-project-frontend/src/app/services/user-type.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { UserType } from '../models/user-type.model';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class UserTypeService {
  private api = 'http://localhost:8080/api/user-types';

  constructor(private http: HttpClient) {}

  getAll(): Observable<UserType[]> {
    return this.http.get<UserType[]>(this.api);
  }

  create(type: UserType): Observable<UserType> {
    return this.http.post<UserType>(this.api, type);
  }

  update(id: number, type: UserType): Observable<UserType> {
    return this.http.put<UserType>(`${this.api}/${id}`, type);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.api}/${id}`);
  }
}
